### PR TITLE
Fetch Logs infrastructure

### PR DESCRIPTION
**Pull request by andrewballantyne**
Help for #1653

Adds Pod Logs Infrastructure

https://github.com/opendatahub-io/odh-dashboard/assets/8126518/5dbc2bb5-0df3-4288-8a7b-b6273fc0f47b

### CODE REVIEW

File: backend/src/routes/api/k8s/pass-through.ts [hunk 1]
```diff
@@ -16,7 +16,42 @@ type PassThroughData = {
...
+
+export const passThroughText = (
+  fastify: KubeFastifyInstance,
+  request: OauthFastifyRequest,
+  data: PassThroughData,
+): Promise<string> => {
+  return (
+    proxyCall(fastify, request, data)
+      // TODO: is there bad text states that we want to error out on?
+      .then((rawData) => rawData) // noop intentionally until above inquiry is figured out
```
Comment (andrewballantyne) [inline backend/src/routes/api/k8s/pass-through.ts:49]: I'm not going to change this logic atm -- we can see what happens when @manaswinidas does more of the logging work.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/LogsTab.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/LogsTab.tsx:1]: Probably can toss this file -- this is just an example of how to use.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:1]: Supporting code, probably can toss as well.

**Review by andrewballantyne — COMMENTED** [review]
There is also some client side renaming of the NotebookContainer that can be reverted probably -- but seems pretty okay to go ahead with.

**Review by manaswinidas — COMMENTED** [review]
/lgtm works fine for me
Screenshot 2023-08-23 at 7 01 27 PM

**Review by andrewballantyne — COMMENTED** [review]
```
const LogsTab: React.FC<LogsTabProps> = ({ task }) => {
  const { namespace } = usePipelinesAPI();
  const [selectedContainer, setSelectedContainer] = React.useState<PodContainer | null>(null);
  const podName = task.runDetails?.status.podName ?? '';
  const podContainers = usePodContainers(namespace, podName);
  const [logs, loaded, error] = useFetchLogs(podName, selectedContainer?.name ?? '');
```

Here is the code usage ... deleting my UI files as they were test files.

**Comment by andrewballantyne** [comment]
@manaswinidas removed my UI code... let me know if you're still good with this.

/approve

**Comment by andrewballantyne** [comment]
Going to reapply Das' approval -- it'll be nice for her to have the code for tomorrow, her timezone.

**Pull request merged**